pl_examples/full_examples/semantic_segmentation/semseg.py

```diff
@@ -0,0 +1,151 @@
+            return img, mask
+        else :
+            return img
+    
+    def encode_segmap(self, mask):
+        '''
+        Sets void classes to zero so they won't be considered for training
+        '''
+        for voidc in self.void_labels :
+            mask[mask == voidc] = self.ignore_index
+        for validc in self.valid_labels :
+            mask[mask == validc] = self.class_map[validc]
+        return mask
+    
+    def get_filenames(self, path):
+        files_list = list()
+        for filename in os.listdir(path):
+            files_list.append(os.path.join(path, filename))
+        return files_list
+    
+class SegModel(pl.LightningModule):
+    def __init__(self, hparams):
+        super(SegModel, self).__init__()
+        self.hparams = hparams
+        self.root_path = hparams.root
+        self.batch_size = hparams.batch_size
+        self.learning_rate = hparams.lr
+        self.net = torchvision.models.segmentation.fcn_resnet50(pretrained = False, progress = True, num_classes = 19)
+        self.transform = transforms.Compose([
+            transforms.ToTensor(),
+            transforms.Normalize(mean = [0.35675976, 0.37380189, 0.3764753], std = [0.32064945, 0.32098866, 0.32325324])
+        ])
+        self.trainset = KITTI(self.root_path, split = 'train', transform = self.transform)
+        self.testset = KITTI(self.root_path, split = 'test', transform = self.transform)
+        
+    def forward(self, x):
+        return self.net(x)
+    
+    def training_step(self, batch, batch_nb) :
+        img, mask = batch
+        img = img.float()
+        mask = mask.long()
+        out = self.forward(img)
+        loss_val = F.cross_entropy(out['out'], mask, ignore_index = 250)
+        return {'loss' : loss_val}
+    
+#     def test_step(self, batch, batch_nb):
+#         print('-----------------testing-----------------')
+#         img = batch
+#         # log sampled images
+#         masks = self.net(img)
+#         grid = torchvision.utils.make_grid(masks)
+#         self.logger.experiment.add_image(f'generated_images', grid, self.current_epoch)
+    
+    def configure_optimizers(self):
+        opt = torch.optim.Adam(self.net.parameters(), lr = self.learning_rate)
+        sch = torch.optim.lr_scheduler.CosineAnnealingLR(opt, T_max = 10)
+        return [opt], [sch]
+    
+    def train_dataloader(self):
+        return DataLoader(self.trainset, batch_size = self.batch_size, shuffle = True)
+    
+    def test_dataloader(self):
+        return DataLoader(self.testset, batch_size = 1, shuffle = True)
+    
+def main(hparams):
+    # ------------------------
+    # 1 INIT LIGHTNING MODEL
+    # ------------------------
+    model = SegModel(hparams)
+
+    # ------------------------
+    # 2 INIT TRAINER
+    # ------------------------
+    trainer = pl.Trainer(
+        gpus = hparams.gpus
+    )
+
+    # ------------------------
+    # 3 START TRAINING
+    # ------------------------
+    trainer.fit(model)
```

Comment:
It would be nice to also add inference usage, like segment an image from dataset...